pennylane/transforms/mitigate.py

```diff
@@ -12,13 +12,309 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Provides transforms for mitigating quantum circuits."""
+from copy import copy
+
 from typing import Any, Dict, Optional, Sequence, Union
 
-from pennylane import QNode, apply
-from pennylane.math import mean, shape
+from pennylane import QNode, apply, adjoint
+from pennylane.math import mean, shape, round
 from pennylane.tape import QuantumTape
 from pennylane.transforms import batch_transform
 
+import pennylane as qml
+
+
+@batch_transform
+def fold_global(circuit, scale_factor):
+    r"""Differentiable circuit folding of the global unitary ``circuit``.
+
+    For a unitary ``circuit`` :math:`U = L_d .. L_1`, where :math:`L_i` can be either a gate or layer, ``fold_global`` constructs
+
+    .. math:: \text{fold_global}(U) = U (U^\dagger U)^n (L^\dagger_d L^\dagger_{d-1} .. L^\dagger_s) (L_s .. L_d)
+
+    where :math:`n = \lfloor (\lambda - 1)/2 \rfloor` and :math:`s = \lfloor \left((\lambda -1) \mod 2 \right) (d/2) \rfloor` are determined via the ``scale_factor`` :math:`=\lambda`.
```

Comment:
Here you multiply with d/2?

Reply:
yes! dont forget that `fraction_scale` later gets multiplied by `d/2`, see line 216: `num_to_fold = int(round(fraction_scale * n_ops / 2))`. Is that what you meant?